app/src/common/shared/org/mozilla/vrbrowser/SessionStore.java

```diff
@@ -719,6 +724,11 @@ public void onLocationChange(GeckoSession aSession, String aUri) {
         for (GeckoSession.NavigationDelegate listener: mNavigationListeners) {
             listener.onLocationChange(aSession, aUri);
         }
+
+        // The homepage finishes loading after the region has been updated
+        if (mRegion != null && aUri.equalsIgnoreCase(SessionStore.HOME_WITHOUT_REGION_ORIGIN)) {
```

Comment:
you can replace the latter part with `isHomeUri(aUri)`

Reply:
I use equalsIgnoreCase instead of the startsWith from isHomeUri to avoid infinite loading when the #region= is loaded as it also starts with the HOME_WITHOUT_REGION_ORIGIN string.

Reply:
ah, good call

Reply:
could you change `isHomeUri` to something like this:

```diff
diff --git a/app/src/common/shared/org/mozilla/vrbrowser/SessionStore.java b/app/src/common/shared/org/mozilla/vrbrowser/SessionStore.java
index 6d6ea25..9ff011a 100644
--- a/app/src/common/shared/org/mozilla/vrbrowser/SessionStore.java
+++ b/app/src/common/shared/org/mozilla/vrbrowser/SessionStore.java
@@ -432,8 +432,14 @@ public class SessionStore implements GeckoSession.NavigationDelegate, GeckoSessi
         return result;
     }
 
-    public Boolean isHomeUri(String aUri) {
-        return aUri != null && aUri.toLowerCase().startsWith(SessionStore.HOME_WITHOUT_REGION_ORIGIN);
+    public Boolean isHomeUri(String aUri, boolean aExactMatch) {
+        if (aUri == null || aUri.length == 0) {
+            return false;
+        }
+        if (aExactMatch === true) {
+            return aUri.equalsIgnoreCase(SessionStore.HOME_WITHOUT_REGION_ORIGIN);
+        }
+        return aUri.toLowerCase().startsWith(SessionStore.HOME_WITHOUT_REGION_ORIGIN);
     }
 
     public String getCurrentUri() {
```

Reply:
I'm generally against booleans in APIs. When reading the code, it is really hard to understand what the boolean in `isHomeUri("some.uri", false)` means. I would prefer two functions `isHomeUri()` and `startsWithHomeUri()`.

Reply:
yeah, that's better - thanks 👍will change later